### DISCUSSION TITLE
prebuild snapshots

### DISCUSSION THREAD

**Discussion by wepl** [vAmigaWeb/vAmigaWeb#181] [General]
Is there a way to prebuild snapshots?
I like to create such (memory) images which include all data to run them with vAmigaWeb.
Where can I find the necessary info?

**Comment by mithrendal** [comment]
you build snapshots by 
* first boot everything until you reached the state you like to snapshot
* choose `take the snapshot` from menubar 
* choose `snapshot browser` from menubar
* now you see the just taken snapshot
* on the bottom right corner of each snapshot picture is a download symbol
* 
![IMG_0333](https://github.com/vAmigaWeb/vAmigaWeb/assets/17108995/f0ffc125-1b19-4143-b32b-ca649fa1b132)

* hit the download symbol … save it to your local filesystem

now you start vamigaweb by appending the url to that snapshot file like this sample adf file

https://vamigaweb.github.io#https://vamigaweb.github.io/doc/media/SysInfo4.4.adf


for more info on how to preload stuff see the wiki as a starting point
https://github.com/vAmigaWeb/vAmigaWeb/wiki/simple_preconfig

**Reply by wepl** [reply]
Thanks a lot.
I want to assemble snapshots myself. Are there specs of the file format of the snapshots anywhere?

**Comment by dirkwhoffmann** [comment]
A snapshot contains the entire internal state of the emulator. It's a moving target because whenever an internal variable is added, modified, or removed, the snapshot format changes. Hence, snapshots are not interchangeable between releases, and they were never meant to be. The original idea of snapshots was to master difficult games by taking a snapshot from time to time, enabling the user to "rewind" a few seconds.

**Comment by mithrendal** [comment]
you just might do a search on vAmigaWeb for the term `snapshot` which will bring involved classes up especially `Snapshot.cpp` where you can look how the header of the snapshot is setup … but as @dirkwhoffmann said, this is all very proprietary and almost always changes when new variables/features are added to new core versions which the previous snapshot format did not know about. Therefore the state of a newer core can not be restored from an older snapshot file…

**Comment by wepl** [comment]
Thanks I understand. 
It seems the Savestates in vAmiga are less generic than the savestates in WinUAE where they are chunk based and all runtime specific information can be omitted.
Basically my idea was to provide only
- memory setup (which memory should exist)
- cpu type
- chipset type
- memory content
- start pc any some start registers (a0-a7)

I started some years ago to create uss files (WinUAE) but didn't finish this due lack of time.
Now I thought to resurrect this using vAmigaWeb to run in a webbrowser.
